nxapi/nxlog.py

```diff
@@ -77,3 +126,75 @@ def explain_nxlog(nxlog):
 
     return explain
 
+def unify_date(date):
+    """ tries to parse a text date,
+    returns date object or None on error """
+    out_date_format = "%Y%m%dT%H:%M:%S"
+    idx = 0
+    res = ""
+    utc_shift=0
+    success = 0
+    supported_formats = [
+        "%b  %d %H:%M:%S",
+        "%b %d %H:%M:%S",
+        "%Y/%m/%d %H:%M:%S",
+        "%Y-%m-%d %H:%M:%S",
+        "%Y-%m-%dT%H:%M:%S"
+    #            "%Y-%m-%dT%H:%M:%S+%:z"
+        ]
+    # Seems coherent to store UTC time
+    # The RFC 3339 specifies CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm] as a format
+    # This is not RFC 3339 compliant...
+    while date[idx] == " " or date[idx] == "\t":
+        idx += 1
+        success = 0
+    for date_format in supported_formats:
+        # strptime does not support numeric time zone, hack.
+        idx = date.find("+")
+        if idx != -1:
+            utc_shift = date[idx:]
+            date = date[:idx]
+        try:
+            x = datetime.strptime(date, date_format)
+            x = x.replace(hour=x.hour+int(utc_shift))
+            # ugly hack when we don't have year
+            if x.year==1900:
+                x=x.replace(year=datetime.now().year)
+            z = x.strftime(out_date_format)
+            success = 1
+            break
+        except:
+            #print "'"+clean_date+"' not in format '"+date_format+"'"
+            pass
+    if success == 0:
+        logging.critical("Unable to parse date format :'"+date+"'")
+        return ""
+    return z
+
+def coords(ip, db='/usr/share/GeoIP/GeoIPCity.dat'):
+    ret=None
+    try:
+        import GeoIP
+    except:
+        logging.warning("GeoIP python module not installed")
+        return ret
+    try:
+        socket.inet_aton(ip)
+        ip_type=4
+    except socket.error:
+        try:
+            socket.inet_pton(socket.AF_INET6, ip)
+            ip_type=6
+        except socket.error:
+            logging.warning("Ip %s is not parseable" % ip)
+            return None
+    r=None
+    if ip_type==4:
+        gi = GeoIP.open(db, GeoIP.GEOIP_STANDARD)
```

Comment:
I will at least add the caching, but this will not make me happy with this piece of code.